Chrysalis-Papilio/Soen-490-Papilio-Android
# Activity search and filter

# General info

**Issue number**: PAPILIO-12 and PAPILIO-13

**Task description**: 

 - Search Feature (Papilio-12) : A search feature that returns activities based on the search query inputted by the user
 - Filter Feature (Papilio-13): A filter feature that filters out activities that don't match your chosen filter settings.  

# Testing
Testing the Search Feature:

1. Open the App, either log in or sign up, and go to the "Browse" tab on the home page
2. Tap on the search bar at the top of the page and put in whatever query you want, then tap the "Search" icon on the top right.
3. Verify that you can see search results that match your search queries
4. Also verify that sometimes that search turns up nothing because there might be no activity based on your queries (in which case, it would show a message box saying "No Activities Found!")


Testing the Filter feature:
1. Open the app and log in or sign up. You should automatically end up on the home page.
2. On the home page, on the top right, tap on the Filter icon. 
3. Choose whatever filters you want and press SAVE
4. Verify that the activities that show up now match your filter settings
5. Tap on the Filter icon again and press REMOVE ALL FILTERS
6. Verify that the filters are now removed and that you can now see all activities again unfiltered.


# Notes
- The UI and User Flow of both features remains subject to change
- More filter options can be added depending on how the Activity Model expands in the DB (if it ever does)
